raft/src/main/scala/zio/raft/Raft.scala

```diff
@@ -532,29 +533,27 @@ class Raft[A <: Command](
 
   private def applyToStateMachineRule =
     for
-      s <- state.get
+      s <- raftState.get
       _ <- ZIO
         .logDebug(
           s"memberId=${this.memberId} applyToStateMachineRule ${s.commitIndex} ${s.lastApplied}"
         )
         .when(s.commitIndex > s.lastApplied)
       newState <- applyToStateMachine(s)
-      _ <- state.set(newState)
+      _ <- raftState.set(newState)
     yield s != newState
 
   private def takeSnapshotRule =
     for
       (snapshotIndex, snapshotSize) <- snapshotStore.latestSnapshotIndexAndSize
-      s <- state.get
-      shouldTakeSnapshot <- stateMachine.get.map(
-        _.shouldTakeSnapshot(snapshotIndex, snapshotSize, s.commitIndex)
-      )
+      s <- raftState.get
+      shouldTakeSnapshot = stateMachine.shouldTakeSnapshot(snapshotIndex, snapshotSize, s.commitIndex)
       _ <- ZIO.logDebug(
         s"memberId=${this.memberId} takeSnapshotRule $shouldTakeSnapshot $snapshotIndex ${s.commitIndex} $snapshotSize"
       )
       _ <- ZIO.when(shouldTakeSnapshot):
         for
-          stream <- stateMachine.get.map(_.takeSnapshot)
+          stream <- ZIO.succeed(stateMachine.takeSnapshot)
```

Comment:
this is ugly, anyway, it is not needed with scala 3.7. You can just use =

Reply:
yes it is 😅 
I'll upgrade to 3.7 :)